src/components/Sort.js

```diff
@@ -59,6 +83,72 @@ const Sort = () => {
     dispatch(sortButtonModsQuery(modsField))
   }
 
+  //component to show the main file if any.
+  const FileElement = ({referenceCurie}) => {
+    const [referenceFiles, setReferenceFiles] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const dispatch = useDispatch();
+    useEffect(() => {
+      const fetchReferencefiles = async () => {
+      setIsLoading(true);
+      const referencefiles = await axios.get(process.env.REACT_APP_RESTAPI + "/reference/referencefile/show_all/" + referenceCurie);
+        setReferenceFiles(referencefiles.data);
+        setIsLoading(false);
+      }
+      fetchReferencefiles().finally()
+    }, [referenceCurie]);
+
+    let cssDisplayLeft = 'Col-display Col-display-left';
+    if (referenceFiles === null ) {
+      return null;
+    }
+
+    const oktaMod = useSelector(state => state.isLogged.oktaMod);
+    const testerMod = useSelector(state => state.isLogged.testerMod);
+    const oktaDeveloper = useSelector(state => state.isLogged.oktaDeveloper);
+    let accessLevel = oktaMod;
+    if (testerMod !== 'No') { accessLevel = testerMod; }
+    else if (oktaDeveloper) { accessLevel = 'developer'; }
+
+    const copyright_license_open_access = getOpenAccess(referenceCurie);
+    const rowReferencefileElements = []
+    for (const[index, referencefileDict] of referenceFiles.entries()) {
+       if (referencefileDict['file_class'] !== 'main') {continue;}//only display main file
+
+       let is_ok = false;
+       let allowed_mods = [];
+       for (const rfm of referencefileDict['referencefile_mods']) {
+        if (rfm['mod_abbreviation'] !== null) { allowed_mods.push(rfm['mod_abbreviation']); }
+        if (rfm['mod_abbreviation'] === null || rfm['mod_abbreviation'] === accessLevel) { is_ok = true; }
+       }
+       let filename = referencefileDict['display_name'] + '.' + referencefileDict['file_extension'];
+       let referencefileValue = (<div><b>{referencefileDict['file_class']}:&nbsp;</b>{filename} &nbsp;({allowed_mods.join(", ")})</div>);
+       if (copyright_license_open_access === true || accessLevel === 'developer') {
+        is_ok = true;
+       } else if (accessLevel === 'No') {
+         is_ok = false;
+         referencefileValue = (<div><b>{referencefileDict['file_class']}:&nbsp;</b>{filename}</div>);
+       }
+      if (is_ok) {
+        referencefileValue = (<div><b>{referencefileDict['file_class']}:&nbsp;</b><button className='button-to-link' onClick={ () =>
+          dispatch(downloadReferencefile(referencefileDict['referencefile_id'], filename, accessToken))
+        } >{filename}</button></div>); }
+        const referencefileRow = (
+         <div>
+           {referencefileValue}
+        </div>);
+
+       rowReferencefileElements.push( referencefileRow ); }
+      return (
+        <>
+         {isLoading ?
+            <Row key="supplementExpandTogglerRow" className="Row-general">
```

Comment:
This probably shouldn't be in a <Row><Col>  it could just be a span or maybe even just html text.  If it looks okay though, that's fine

Reply:
@FB-ping my previous comment didn't show well because of markup.  Row and Col are probably not needed there, but if it looks okay, then cool.